source/io/module_javaScriptInvoke.js

```diff
@@ -155,158 +155,144 @@
-                writer: Module.eggShell.dataWriteTypedArray,
-                isValidReturnType: function (value) {
-                    return value instanceof Float64Array;
+                Module.eggShell.writeDouble(valueRef, data.userValue ? 1 : 0);
+            };
+
+            var visitNumeric = function (valueRef, data) {
+                // TODO-san anything to do for this?
+                // Looks like the data-grid ends up pushing the string value "-Infinity"
+                // instead of the numeric value -Infinity to the model so perform parseFloat
+                // in-case a string is passed instead of a number
+                if (typeof data.userValue !== 'number') {
+                    mergeNewError(data.errorValueRef, data.functionName, ERRORS.kNITypeMismatchForReturnTypeInJavaScriptInvoke);
+                    return;
                 }
-            },
-            JavaScriptRefNum: {
-                reader: dataReadJavaScriptRefNum,
-                writer: dataWriteJavaScriptRefNum,
-                isValidReturnType: function () {
-                    return true;
+                var dataNum = parseFloat(data.userValue);
+                Module.eggShell.writeDouble(valueRef, dataNum);
+            };
+
+            proto.visitInt8 = visitNumeric;
+            proto.visitInt16 = visitNumeric;
+            proto.visitInt32 = visitNumeric;
+            proto.visitUInt8 = visitNumeric;
+            proto.visitUInt16 = visitNumeric;
+            proto.visitUInt32 = visitNumeric;
+            proto.visitSingle = visitNumeric;
+            proto.visitDouble = visitNumeric;
+            proto.visitEnum8 = visitNumeric;
+            proto.visitEnum16 = visitNumeric;
+            proto.visitEnum32 = visitNumeric;
+
+            proto.visitString = function (valueRef, data) {
+                if (typeof data.userValue !== 'string') {
+                    mergeNewError(data.errorValueRef, data.functionName, ERRORS.kNITypeMismatchForReturnTypeInJavaScriptInvoke);
+                    return;
                 }
-            }
-        };
+                Module.eggShell.writeString(valueRef, data.userValue);
+            };
 
-        var createJavaScriptParametersArray = function (parametersPointer, parametersCount) {
-            var parameters = new Array(parametersCount);
-            for (var index = 0; index < parametersCount; index += 1) {
-                var typeName = getParameterTypeString(parametersPointer, index);
-                var parameterPointer = JavaScriptInvoke_GetParameterPointer(parametersPointer, index);
-                if (typeName === 'Array') {
-                    typeName += getArrayElementTypeString(parameterPointer);
+            proto.visitArray = function (valueRef, data) {
+                try {
+                    Module.eggShell.resizeArray(valueRef, [data.userValue.length]);
+                    Module.eggShell.writeTypedArray(valueRef, data.userValue);
+                    return;
```

Comment:
unneccessary return? I guess just make it consistent with visitJSObjectRefnum

Reply:
Done.